config/config.go

```diff
@@ -92,6 +113,71 @@ func Unmarshall(flowConfig *FlowConfig) error {
 	return nil
 }
 
+// Print prints current configuration keys and values.
+// Returns:
+// map[string]struct{}: map of keys to avoid printing if they were set by an config file.
+// This is required because we still have other config values not migrated to the config package. When a
+// config file is used currently only network-configs are set, we want to avoid printing the config file
+// value and also the flag value.
+func Print(info *zerolog.Event, flags *pflag.FlagSet) map[string]struct{} {
+	// only print config values if they were overridden with a config file
+	m := make(map[string]struct{})
+	if flags.Lookup(configFilePath).Changed {
+		for _, key := range conf.AllKeys() {
+			info.Str(key, fmt.Sprintf("%v", conf.Get(key)))
+			s := strings.Split(key, ".")
+			if len(s) == 2 {
+				m[s[1]] = struct{}{}
+			} else {
+				m[key] = struct{}{}
+			}
+		}
+	}
+
+	return m
+}
+
+// setAliases sets aliases for config sub packages. This should be done directly after pflags are bound to the configuration store.
+// Upon initialization the conf will be loaded with the default config values, those values are then used as the default values for
+// all the CLI flags, the CLI flags are then bound to the configuration store and at this point all aliases should be set if configuration
+// keys do not match the CLI flags 1:1. ie: networking-connection-pruning -> network-config.networking-connection-pruning. After aliases
+// are set the conf store will override values with any CLI flag values that are set as expected.
+func setAliases() {
+	err := network.SetAliases(conf)
+	if err != nil {
+		panic(fmt.Errorf("failed to set network aliases: %w", err))
+	}
+}
+
+// overrideConfigFile overrides the default config file by reading in the config file at the path set
+// by the --config-file and --config-file-name flags in our viper config store.
+//
+// Returns:
+//
+//	error: if there is any error encountered while reading new config file, all errors are considered irrecoverable.
+//	bool: true if the config was overridden by the new config file, false otherwise or if an error is encountered reading the new config file.
+func overrideConfigFile(flags *pflag.FlagSet) (error, bool) {
```

Comment:
The function `overrideConfigFile` returns `(error, bool)`. It is more idiomatic in Go to have the error as the last return value. So it is better to have it return `(bool, error)`.

Reply:
https://github.com/onflow/flow-go/pull/4340/commits/6e8c1a1ddfedcb4b68d8b3fd320a2e9e547266f7